From 5f9289f1dc08762fe61777a5a9e6e27b182b2380 Mon Sep 17 00:00:00 2001
From: Alexandre Bailon <abailon@baylibre.com>
Date: Tue, 22 Mar 2022 09:31:49 +0000
Subject: [PATCH 10/10] Fix a wrong implicit type cast

NNAPI fails to build because of an unknown symbol:
bool android::nn::validateRequest<android::hardware::neuralnetworks::V1_0::Request, android::hardware::neuralnetworks::V1_3::Model>(android::hardware::neuralnetworks::V1_0::Request const&, android::hardware::neuralnetworks::V1_3::Model const&, bool)
There is an implicit type casting that cast to a wrong type and causes
the issue. This explicitly convert the variable to avoid the implicit
caste and fix the build issue.

Signed-off-by: Alexandre Bailon <abailon@baylibre.com>
---
 ArmnnPreparedModel.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ArmnnPreparedModel.cpp b/ArmnnPreparedModel.cpp
index d87f9f8..fea9048 100644
--- a/ArmnnPreparedModel.cpp
+++ b/ArmnnPreparedModel.cpp
@@ -191,7 +191,7 @@ Return<V1_0::ErrorStatus> ArmnnPreparedModel<HalVersion>::execute(
         return V1_0::ErrorStatus::INVALID_ARGUMENT;
     }
 
-    if (!android::nn::validateRequest(request, m_Model))
+    if (!android::nn::validateRequest(request, convertToV1_0(m_Model)))
     {
         NotifyCallbackAndCheck(callback, V1_0::ErrorStatus::INVALID_ARGUMENT, "ArmnnPreparedModel::execute");
         return V1_0::ErrorStatus::INVALID_ARGUMENT;
-- 
2.18.0

